### PR TITLE
feat(jira): Support outbound status sync

### PR DESCRIPTION
**Pull request by macqueen**
diff is really large, but a lot of it is sample API responses from Jira

### CODE REVIEW

File: src/sentry/integrations/jira/integration.py [hunk 1]
```diff
@@ -505,12 +506,58 @@ def sync_assignee_outbound(self, external_issue, user, assign=True, **kwargs):
...
+        except IndexError:
+            # TODO(jess): Email for failure
+            logger.warning(
+                'jira.status-sync-fail',
+                extra={
+                    'organization_id': external_issue.organization_id,
+                    'integration_id': external_issue.integration_id,
+                    'issue_key': external_issue.key,
+                }
+            )
```
Comment (mnoble) [inline src/sentry/integrations/jira/integration.py]: Should we return or something here or do we want it to actually explode on the next line (`transition['id']`)?

Comment (macqueen) [reply src/sentry/integrations/jira/integration.py]: woops no, meant to return

Comment (macqueen) [reply src/sentry/integrations/jira/integration.py]: fixed

**Review by mnoble — APPROVED** [review]
Looks good.

I'm guessing `sync_status_outbound` (and methods like it) are only ever called from a background job and not anywhere that needs to react to the result?

**Comment by macqueen** [comment]
@mnoble yup, it gets called from an async task: https://github.com/getsentry/sentry/blob/master/src/sentry/tasks/integrations.py#L79

I'd like to add some sort of email functionality where we'd just tell users their issue failed to sync (which is what we do for commit fetching, for example) and I have a ticket open for that.

**Pull request merged**